src/tools/annotation/bidirectionalTool/utils/updatePerpendicularLineHandles.js

```diff
@@ -20,24 +35,28 @@ export default function(eventData, data) {
       y: (start.y + end.y) / 2,
     };
 
-    // Length of long-axis
-    const dx = (start.x - end.x) * (eventData.image.columnPixelSpacing || 1);
-    const dy = (start.y - end.y) * (eventData.image.rowPixelSpacing || 1);
-    const length = Math.sqrt(dx * dx + dy * dy);
-
-    const vectorX = (start.x - end.x) / length;
-    const vectorY = (start.y - end.y) / length;
-
-    const perpendicularLineLength = length / 2;
-
-    startX = mid.x + (perpendicularLineLength / 2) * vectorY;
-    startY = mid.y - (perpendicularLineLength / 2) * vectorX;
-    endX = mid.x - (perpendicularLineLength / 2) * vectorY;
-    endY = mid.y + (perpendicularLineLength / 2) * vectorX;
+    // Inclination of the perpendicular line
+    const vector = getLineVector(
+      columnPixelSpacing,
+      rowPixelSpacing,
+      start,
+      end
+    );
+
+    const perpendicularLineLength = vector.length / 2;
+    const rowMultiplier = perpendicularLineLength / (2 * rowPixelSpacing);
+    const columnMultiplier = perpendicularLineLength / (2 * columnPixelSpacing);
+
+    startX = mid.x + columnMultiplier * vector.y;
+    startY = mid.y - rowMultiplier * vector.x;
+    endX = mid.x - columnMultiplier * vector.y;
+    endY = mid.y + rowMultiplier * vector.x;
   }
 
-  data.handles.perpendicularStart.x = startX;
-  data.handles.perpendicularStart.y = startY;
-  data.handles.perpendicularEnd.x = endX;
-  data.handles.perpendicularEnd.y = endY;
+  measurementData.handles.perpendicularStart.x = startX;
+  measurementData.handles.perpendicularStart.y = startY;
+  measurementData.handles.perpendicularEnd.x = endX;
+  measurementData.handles.perpendicularEnd.y = endY;
```

Comment:
The same again.
I don't want to touch upper levels of the tool code structure.
It might have some undesired impacts and would add a lot of effort for me to test everything again.